document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -0,0 +1,347 @@
+          break;
+        case GT:
+          filterString.append(" > ");
+          break;
+        case LT:
+          filterString.append(" < ");
+          break;
+        case GTE:
+          filterString.append(" >= ");
+          break;
+        case LTE:
+          filterString.append(" <= ");
+          break;
+        case EXISTS:
+          // TODO: Checks if key exists
+          break;
+        case NOT_EXISTS:
+          // TODO: Checks if key does not exist
+          break;
+        case AND:
+        case NEQ:
+        case OR:
+          throw new UnsupportedOperationException("Only Equality predicate is supported");
+        default:
+          break;
+      }
+      return filterString.append("'" + value + "'").toString();
+    }
+  }
+  
+  /**
+   * If the query column is a json column, then add document column prefix to it\
+   * This will currently only search first level keys inside JSON
+   * TODO: Add support for deep nested keys and arrays
+   */
+  private String getFieldPrefix(String fieldName) {
+    String fieldPrefix = fieldName;
+    if (!OUTER_COLUMNS.contains(fieldName)) {
+      fieldPrefix = DOCUMENT_KEY + "->>" + "'" + fieldName + "'";
+    }
+    return fieldPrefix;
+  }
+  
+  private String parseOrderByQuery(List<OrderBy> orderBys) {
+    String orderBySQL = orderBys
+      .stream()
+      .map(orderBy -> orderBy.getField() + " " + (orderBy.isAsc() ? "ASC" : "DESC"))
+      .filter(str -> !str.isEmpty())
+      .collect(Collectors.joining(" , "));
+    
+    return orderBySQL;
+  }
+  
+  @Override
+  public boolean delete(Key key) {
+    String deleteSQL = String.format("DELETE FROM %s WHERE %s = ?", collectionName, ID_KEY);
+    try {
+      PreparedStatement preparedStatement = client.prepareStatement(deleteSQL);
+      preparedStatement.setString(1, key.toString());
+      preparedStatement.executeUpdate();
+      return true;
+    } catch (SQLException e) {
+      LOGGER.error("SQLException deleting document. key: {}", key, e);
+    }
+    return false;
+  }
+  
+  @Override
+  public boolean deleteSubDoc(Key key, String subDocPath) {
+    return false;
+  }
+  
+  @Override
+  public boolean deleteAll() {
+    String deleteSQL = String.format("DELETE FROM %s", collectionName);
+    try {
+      PreparedStatement preparedStatement = client.prepareStatement(deleteSQL);
+      preparedStatement.executeUpdate();
+      return true;
+    } catch (SQLException e) {
+      LOGGER.error("SQLException deleting all documents.", e);
+    }
+    return false;
+  }
+  
+  @Override
+  public long count() {
+    String countSQL = String.format("SELECT COUNT(*) FROM %s", collectionName);
+    try {
+      PreparedStatement preparedStatement = client.prepareStatement(countSQL);
+      ResultSet resultSet = preparedStatement.executeQuery();
+      return resultSet.getLong(1);
+    } catch (SQLException e) {
+      LOGGER.error("SQLException deleting all documents.", e);
+    }
+    return 0;
```

Comment:
Interesting!! One more case, I guess need to correct the contract.
Q: Can we go with `-1` as an error case?